src/components/export.tsx

```diff
@@ -1,51 +1,214 @@
+  /**
+   * Send the request for export
+   * @param option selected export option
+   * @returns Downloads the selected file
+   */
+  const startExporting = (option: any) => () => {
+    const formatType = option.type;
+
+    switch(formatType) {
+      case 'DIRECT':
+        location.href = reference.csvDownloadLink;
+        break;
+      case 'BAG':
+      case 'FILE': 
+        setIsDownloading(true);
+        setShowProgressModal(true);
+        setSelectedOption(option);
+        const bagName = reference.table.name;
+        const exporter = new ConfigService.ERMrest.Exporter(
+          reference,
+          bagName,
+          option,
+          ConfigService.chaiseConfig.exportServicePath
+        );
+        const exportParametersString = JSON.stringify(exporter.exportParameters, null, '  ');
+
+        // begin export and start a timer
+        console.info('Executing external export with the following parameters:\n' + exportParametersString);
+        console.time('External export duration');
+
+        setExporterObj(exporter);
+        if (exporter) {
+          const logStack = LogService.addExtraInfoToStack(null, {
+              template: {
+                  displayname: exporter.template.displayname,
+                  type: exporter.template.type
+              }
+          });
+          const logObj = {
+              action: LogService.getActionString(LogActions.EXPORT),
+              stack: logStack
+          }
+          exporter
+            .run(logObj)
+            .then((response: any) => {
+              setShowProgressModal(false);
+              setIsDownloading(false);
+
+              console.timeEnd('External export duration');
+
+              // if it was canceled, just ignore the result
+              if (response.canceled) return;
+              location.href = response.data[0];
+            })
+            .catch((error: any) => {
+              setShowProgressModal(false);
+              setIsDownloading(false);
 
-    return <span className='chaise-btn-icon fa-solid fa-file-export' />;
+              console.timeEnd('External export duration');
+              
+              error.subMessage = error.message;
+              error.message = 'Export failed. Please report this problem to your system administrators.';
+              
+              dispatchError({ 
+                error: error,
+                isGlobal: true 
+              });
+            });
+        }
+        break;
+      default:
+        dispatchError({ 
+          error: new Error('Unsupported export format: ' + formatType + '. Please report this problem to your system administrators.'),
+          isGlobal: true 
+        });
+    }
+  };
+
+  const cancelExport = () => {
+    // Cancel download
+    if (!!exporterObj) {
+      exporterObj.cancel();
+    }
   }
 
+  const closeModal = () => {
+    cancelExport();
+    setShowProgressModal(false);
```

Comment:
Assuming you remove all the unnecessary state variables and only set the `selectedOption`, then in here you would clear it.

This would also fix a bug in your current implementation where the button stays disabled